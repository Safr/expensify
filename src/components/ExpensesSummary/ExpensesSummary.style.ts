import styled from 'styled-components';
import { Link } from 'react-router-dom';
import { blue, fontSizeLarge, fontSizeSmall, lSize, mSize, offWhite, sSize, white } from '../../theme/variables';
import media from '../../theme/media';

export const StyledLink = styled(Link)`
  display: inline-block;
  padding: ${sSize};
  margin-top: ${mSize};
  margin-right: auto;
  font-size: ${fontSizeSmall};
  font-weight: 300;
  line-height: 1;
  color: ${white};
  background: ${blue};
  border: none;
  ${media.tablet`
    font-size: ${fontSizeLarge};
  `}
`;

export const StyledSpan = styled.span`
  font-weight: 700;
`;

export const PageHeader = styled.div`
  padding: ${sSize};
  font-size: ${fontSizeSmall};
  background-color: ${offWhite};
  ${media.tablet`
    padding: ${lSize};
    font-size: ${fontSizeLarge};
  `}

  h2 {
    text-align: left;
  }
`;
