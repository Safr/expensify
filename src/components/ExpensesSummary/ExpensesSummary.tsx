import React from 'react';
import { connect } from 'react-redux';
import numeral from 'numeral';
import selectExpenses from '../../selectors/expenses';
import selectTotalExpenses from '../../selectors/expensesTotal';
import Container from '../../theme/grid';
import StoreState from '../../types';
import { PageHeader, StyledLink, StyledSpan } from './ExpensesSummary.style';

interface Props {
  expenseCount: number;
  expensesTotal: number;
}

export const ExpensesSummary: React.SFC<Props> = ({ expenseCount, expensesTotal }) => {
  const expenseWord = expenseCount === 1 ? 'expense' : 'expenses';
  const formattedExpenseTotal = numeral(expensesTotal / 100).format('$0,0.00');
  return (
    <PageHeader>
      <Container>
        <h2>
      Viewing <StyledSpan>{expenseCount}</StyledSpan> {expenseWord}
          {' '} totalling <StyledSpan>{formattedExpenseTotal}</StyledSpan>
        </h2>
        <StyledLink to="/create">Add Expense</StyledLink>
      </Container>
    </PageHeader>

  );
};

const mapStateToProps = (state: StoreState) => ({
  expenseCount: selectExpenses(state.expenses, state.filters).length,
  expensesTotal: selectTotalExpenses(selectExpenses(state.expenses, state.filters)),
});

export default connect(mapStateToProps)(ExpensesSummary);
