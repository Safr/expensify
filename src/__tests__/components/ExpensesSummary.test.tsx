import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary/ExpensesSummary';

describe('ExpensesSummary Component', () => {
  test('should render ExpensesSummary correctly with 1 expense', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={222} />);
    expect(wrapper).toMatchSnapshot();
  });
  test('should render ExpensesSummary correctly with multiple expenses', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={33} expensesTotal={22222222} />);
    expect(wrapper).toMatchSnapshot();
  });
});
