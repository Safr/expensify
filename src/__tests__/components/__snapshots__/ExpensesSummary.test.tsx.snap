// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExpensesSummary Component should render ExpensesSummary correctly with 1 expense 1`] = `
<styled.div>
  <styled.div>
    <h2>
      Viewing 
      <styled.span>
        1
      </styled.span>
       
      expense
       
       totalling 
      <styled.span>
        $2.22
      </styled.span>
    </h2>
    <Styled(Link)
      to="/create"
    >
      Add Expense
    </Styled(Link)>
  </styled.div>
</styled.div>
`;

exports[`ExpensesSummary Component should render ExpensesSummary correctly with multiple expenses 1`] = `
<styled.div>
  <styled.div>
    <h2>
      Viewing 
      <styled.span>
        33
      </styled.span>
       
      expenses
       
       totalling 
      <styled.span>
        $222,222.22
      </styled.span>
    </h2>
    <Styled(Link)
      to="/create"
    >
      Add Expense
    </Styled(Link)>
  </styled.div>
</styled.div>
`;
